### PR TITLE
feat(metric): add metric stale SSTs count

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add metric for count of stale SSTs in object store that is no longer referenced by LSMtree.
This metric is expected to be 0 for most of the time. It should only accumulate when SSTs deletion is lagging behind, either due to no worker or some object store connection failure.

Note that 

- Number of orphan SSTs (never referenced by LSMtree) is not counted in this metric. It need be obtained by scanning object store (currently only occur during full GC).
- Metric for total size of stale SSTs is not implemented. Because their file size is not always available:
  - delta log contains `SstableInfo` for added SSTs and only SST id for removed SSTs.
  - SST file size is available in delta log's added SSTs' `SstableInfo`.
  - delta log that contains **only** added SSTs may have been deleted when calculating this total size metric.
- To support the total size metric, we need store at least file size for delta log's removed SSTs. Shall we do it?

## Checklist

- [x] I have written necessary rustdoc comments
~~- [ ] I have added necessary unit tests and integration tests~~
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/storage/src/hummock/sstable_store.rs [hunk 1]
```diff
@@ -375,7 +375,7 @@ impl SstableStore {
 
     pub async fn list_ssts_from_object_store(&self) -> HummockResult<Vec<ObjectMetadata>> {
         self.store
-            .list(&self.path)
```
Comment (Li0k) [inline src/storage/src/hummock/sstable_store.rs:378]: Shall we need to add some formatting checks to the `path`? Adding "/" directly is not necessarily the right thing to do. We can check it at path initialization and use the same rule here to perform assert here

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -1759,6 +1761,7 @@ where
             Excluded(old_checkpoint_id),
             Included(new_checkpoint_id),
         ));
+        trigger_stale_ssts_stat(&self.metrics, versioning.ssts_to_delete.len());
```
Comment (Li0k) [inline src/meta/src/hummock/manager/mod.rs:1764]: In this pr, we need to `trigger_stale_ssts_stat` after changing the `versioning.ssts_to_delete`,This call is easily missed after we add new logic.
Can we make versioning maintain metrics, and wrap ssts_to_delete with metrics reporting into a function to alleviate the above problem?

File: src/storage/src/hummock/sstable_store.rs (hunk 1 above)
Comment (zwang28) [reply src/storage/src/hummock/sstable_store.rs:378]: We should validate path. Will leave a TODO.

Comment (zwang28) [reply src/storage/src/hummock/sstable_store.rs:378]: This added "/" always matches exactly the same "/" added in [get_sst_data_path](https://github.com/risingwavelabs/risingwave/blob/f309b246571a9ec68201d696fbcd5cfd4181519f/src/storage/src/hummock/sstable_store.rs#L288). It aims to filter out path with same prefix.

**Comment by hzxa21** [comment]
> To support the total size metric, we need store at least file size for delta log's removed SSTs. Shall we do it?

Can we infer the total size of removed SSTs (including orphan SSTs) from `object store usage - total size of SSTs in the current version`?

**Review by hzxa21 — APPROVED** [review]
LGTM!

**Review by Li0k — APPROVED** [review]

**Comment by zwang28** [comment]
> Can we infer the total size of removed SSTs (including orphan SSTs) from object store usage - total size of SSTs in the current version?

The total size of object store usage is hard to know: though s3 tells bucket storage size, but we sometimes share one bucket between tenants. There seems no cheap way to tell total prefix storage size.

**Comment by hzxa21** [comment]
> > Can we infer the total size of removed SSTs (including orphan SSTs) from object store usage - total size of SSTs in the current version?
> 
> The total size of object store usage of one tenant is hard to know: though s3 tells bucket storage size, but we sometimes share one bucket between tenants. There seems no cheap way to tell total prefix storage size.

That makes sense. Per prefix monioring in S3 charges extra money

**Pull request merged**